Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -787,6 +787,16 @@ open class NavigationMapView: UIView {
     func setupUserLocation() {
         if !locationManager.isAuthorized() { return }
         
+        // Since Mapbox Maps will not provide location data in case if `LocationOptions.puckType` is
+        // set to nil, we have to draw empty and transparent `UIImage` instead of puck. This is used
+        // in case when user wants to stop showing location puck or draw a custom one.
+        let clearImage = UIColor.clear.image(CGSize(size: 1.0))
```

Comment:
Should we rather create this image only if `userLocationStyle` is `courseView`?

Reply:
We need to use `emptyPuckConfiguration` in other cases as well. Code was updated.